### PR TITLE
test: Add NFS Client deps to Debian etc

### PR DESCRIPTION
**Pull request by mvollmer**
* [x] image-refresh ubuntu-1604
 * [x] image-refresh debian-testing
 * [x] fix check-packagekit failure on debian-testing due to failure to purge unattended-upgrades
 * [x] investigate/fix `Error creating RAID array:The function 'bd_md_create' called, but not implemented!` failures on debian-testing
 * [x] add known issue for new udisks crash (#7854)

### CODE REVIEW

**Comment by martinpitt** [comment]
Why python3-dbus? Is the code itself using that, or just for the tests?

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-4w9cr.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-7812-20171004-145959/

**Comment by cockpituous** [comment]
image-refresh ubuntu-1604 done: https://github.com/cockpituous/cockpit/commits/image-refresh-ubuntu-1604-20171004-153838

**Comment by mvollmer** [comment]
> Why python3-dbus? Is the code itself using that, or just for the tests?

The code is using this.  It talks to the systemd D-Bus API to get the service/session for a PID.

**Comment by martinpitt** [comment]
Apparently NFS and NetworkManager don't work together on Ubuntu 16.04:
```
systemd[1]: NetworkManager.service: Found ordering cycle on NetworkManager.service/verify-active
systemd[1]: NetworkManager.service: Found dependency on dbus.socket/start
systemd[1]: NetworkManager.service: Found dependency on sysinit.target/start
systemd[1]: NetworkManager.service: Found dependency on selinux-basics.service/start
systemd[1]: NetworkManager.service: Found dependency on remote-fs.target/start
systemd[1]: NetworkManager.service: Found dependency on remote-fs-pre.target/start
systemd[1]: NetworkManager.service: Found dependency on nfs-server.service/start
systemd[1]: NetworkManager.service: Found dependency on network.target/start
systemd[1]: NetworkManager.service: Found dependency on NetworkManager.service/verify-active
systemd[1]: NetworkManager.service: Breaking ordering cycle by deleting job dbus.socket/start
```

NFS starts "early" (`sysinit.target`) in the boot sequence to support /usr or /var on NFS, but NM cannot start that early as it requires dbus (it only starts in `multi-user.target`). That was apparently fixed in later OS releases; can we skip the tests on Ubuntu 16.04 for the time being? Would be nice to report this as a downstream bug, it should be a relatively simple fix.

**Comment by martinpitt** [comment]
pitti | mvollmer: search for `case "$RELEASE" in` in the shared `debian.setup`
pitti | mvollmer: there's already special cases for particular releases, just add a `xenial)` one and something like `COCKPIT_DEPS="${COCKPIT_DEPS/nfs-server /}"`

**Comment by mvollmer** [comment]
Let's make a ubuntu-1604 image anyway, just to check that debian.setup is not broken.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-vxjbl.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-7812-20171006-082424/

**Comment by cockpituous** [comment]
image-refresh ubuntu-1604 done: https://github.com/cockpituous/cockpit/commits/image-refresh-ubuntu-1604-20171006-085555

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-mlrl0.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-7812-20171006-100808/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-v0clp.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-7812-20171006-115353/

**Comment by cockpituous** [comment]
image-refresh ubuntu-1604 done: https://github.com/cockpituous/cockpit/commits/image-refresh-ubuntu-1604-20171006-122828

**Comment by martinpitt** [comment]
There is some bug in the tests which breaks debian-testing:
https://fedorapeople.org/groups/cockpit/logs/pull-7812-20171006-144343-3e0aa290-verify-debian-testing/log.html#5

    E: Unable to locate package unattended-upgrades

I'm not entirely sure why that didn't fail before, but this needs to be robustified. I was planning to add unattended-upgrades to our images anyway, but no need to block on that.

**Comment by martinpitt** [comment]
Also, several tests on debian-testing fail like this:

    Error creating RAID array:The function 'bd_md_create' called, but not implemented!

This smells like some new version of the kernel or mdadm or similar, and thus an actual regression somewhere.

**Comment by martinpitt** [comment]
@mvollmer : I pushed the check-packagekit fix (apparently new apt behaviour on latest  debian-testing) on top, and also force-pushed to add a Closes # to your commit (for easier merging).

**Comment by mvollmer** [comment]
`bd_md_create` smells like libblockdev.

**Comment by martinpitt** [comment]
@mvollmer: libblockdev has a new upstream release 2.13, but debian-testing and fedora 27 ship the same version 2.12. There are no changes related to that in in `git diff 2.12-1..  | grep bd_md_create`

**Comment by martinpitt** [comment]
@mvollmer: the difference is that libblockdev-mdraid is missing on debian-testing, while it's present on fedora-27. Our current debian-testing image still has udisks 2.6.5 (before the libblockdev change).

**Comment by martinpitt** [comment]
I pushed two more commits, one for adding libblockdev-mdraid2 and one cleanup. Rebuilding debian-testing for that.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-rgm3v.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-7812-20171010-063232/

**Comment by cockpituous** [comment]
image-refresh debian-testing done: https://github.com/cockpituous/cockpit/commits/image-refresh-debian-testing-20171010-070707

File: bots/images/scripts/debian.setup [hunk 1]
```diff
@@ -71,6 +71,7 @@ case "$RELEASE" in
         # https://tracker.debian.org/pcp not currently in testing
         COCKPIT_DEPS="${COCKPIT_DEPS/libpcp3 /}"
         COCKPIT_DEPS="${COCKPIT_DEPS/pcp /}"
+        TEST_PACKAGES="${TEST_PACKAGES} libblockdev-mdraid2"
```
Comment (mvollmer) [inline bots/images/scripts/debian.setup]: libblockdev-mdraid2 is a run-time dependency of Cockpit, no?  Without it, the mdraid features of Cockpit are broken, and mdraid isn't optional (yet) in the UI.  Also, mdraid is not a plugin for udisks2, right?

So shouldn't some package depend on liblockdev-mdraid2 as well?  Maybe cockit-storaged, maybe udisks2.

Comment (martinpitt) [reply bots/images/scripts/debian.setup]: udisks2 itself doesn't need the mdraid plugin; the dependency was recently demoted from Recommends: to Suggests:, see http://bugs.debian.org/873706 .

If cockpit-storaged requires it, then it should depend on libblockdev-mdraid2. I'll adjust the dependencies accordingly.

Comment (martinpitt) [reply bots/images/scripts/debian.setup]: Updated, please have a look?

**Comment by martinpitt** [comment]
There is still one persistent failure on debian-testing:
```
# testNotRemovingDisks (check_storage_mdraid.TestStorage)
#
Warning: Permanently added '[127.0.0.2]:2301' (ECDSA) to the list of known hosts.
Traceback (most recent call last):
  File "/build/cockpit/bots/../test/verify/check-storage-mdraid", line 284, in testNotRemovingDisks
    b.wait_in_text(info_field("State"), "Running")
  File "/build/cockpit/test/common/testlib.py", line 257, in wait_in_text
    return self.wait_js_func('ph_in_text', selector, text)
  File "/build/cockpit/test/common/testlib.py", line 224, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/common/testlib.py", line 825, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/common/testlib.py", line 851, in _invoke
    raise Error(res['error'])
Error: timeout


not ok 89 testNotRemovingDisks (check_storage_mdraid.TestStorage) # duration: 141s
Error: Could not stat device /dev/md127 - No such file or directory.

NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sda      8:0    0   50M  0 disk 
sdb      8:16   0   50M  0 disk 
sdc      8:32   0   50M  0 disk 
sr0     11:0    1  366K  0 rom  
vda    254:0    0    8G  0 disk 
`-vda1 254:1    0    8G  0 part /
```
The screen shot has  some message about "message recipient disconnected from bus", which sounds like an udisks crash. And indeed the [journal](https://fedorapeople.org/groups/cockpit/logs/pull-7812-20171010-120606-cfb85182-verify-debian-testing/TestStorage-testNotRemovingDisks-debian-testing-127.0.0.2-2301-FAIL.log) has a corresponding stack trace. I. e. upstream bug + knownissue?

**Comment by mvollmer** [comment]
> I. e. upstream bug + knownissue?

Sounds reasonable.

**Comment by martinpitt** [comment]
Filed the udisks crash as known issue 7854 and upstream at https://github.com/storaged-project/udisks/issues/422 and added a commit to include that naughty override.

**Comment by martinpitt** [comment]
Setting priority, we have to work increasingly hard against the bots to not update the images independently, and debian-testing gets out of date quickly.

**Comment by martinpitt** [comment]
The same crash affects `testRaid (check_storage_mdraid.TestStorage)` too sometimes, so this crash is a race condition. I widened the naughty override accordingly.

**Comment by mvollmer** [comment]
> Green at last!

Hurray!  Thanks!

**Pull request merged**